### ISSUE TITLE
Parent scope seems to be wrong

### ISSUE THREAD

**Issue by tailhook** [sunng87/handlebars-rust#93]
The template is:
```
{{# each departments }}
    Department: {{ name }}
    Employees:
        {{# each employees }}
            {{ name }} ({{ ../salary }})
        {{/ each }}
{{/ each }}
```
And the json:
```
{
    "departments": [{
        "name": "developers",
        "salary": 100,
        "employees": [{
            "name": "Jonh Smith"
        }]
    }]
}
```

In http://tryhandlebarsjs.com/ I get the following output:
```
    Department: developers
    Employees:
            Jonh Smith (100)
```

But with this library I get the following output:
```

    Department: developers
    Employees:
        
            Jonh Smith ()
```

As far as I can observe it skips one scope. I.e. I can access `../departments` from the internal `{{#each employees}}` loop (but I would expect it to be `../../departments`). And I can't access the scope of the `{{#each departments}}` loop (so the `../salary` is empty)

**Comment by metadave** [comment]
@tailhook I had ta similar issue in https://github.com/sunng87/handlebars-rust/pull/89.

**Comment by sunng87** [comment]
This should be a different issue introduced in 0.19. I'm working on a fix.

**Comment by sunng87** [comment]
This should be fixed in baf077b . I just added a test case for this.

**Comment by sunng87** [comment]
Thank you @tailhook @metadave , I'm closing this and feel free to reopen if there's any question.
